src/components/Topology/Topology.tsx

```diff
@@ -166,7 +169,7 @@ const Topology = memo(
 
     // Apply deltas to the annotations.
     for (const appName in annotationData) {
-      const annotation = { ...annotationData[appName] };
```

Comment:
Because this was spread it wasn't updating by reference, so the following updates to the annotations were lost.